src/linker/Linker/MarkingHelpers.cs

```diff
@@ -35,7 +35,9 @@ public void MarkForwardedScope (TypeReference typeReference)
 
 			if (typeReference.Scope is AssemblyNameReference) {
 				var assembly = _context.Resolve (typeReference.Scope);
-				if (assembly != null && assembly.MainModule.GetMatchingExportedType (_context.TryResolve (typeReference), out var exportedType))
+				if (assembly != null &&
+					_context.TryResolve (typeReference) is TypeDefinition typeDefinition &&
```

Comment:
I think we already check for nullability in `GetMatchingExportedType`.

Reply:
I somewhat prefer to annotate the input parameter to `GetMatchingExportedType` as non-nullable, and require callers to decide what to do if a type doesn't resolve - LMK if you prefer making it nullable instead.

Reply:
> I somewhat prefer to annotate the input parameter to `GetMatchingExportedType` as non-nullable, and require callers to decide what to do if a type doesn't resolve

Makes sense. We can keep it this way :)